### PR TITLE
Bugfix/include what you use

### LINKED ISSUE #1485: include-what-you-use errors
## Description

The following files (at least) have include-what-you-use violations:

* `rev/mat/fun/ordered_constrain.hpp`
* `rev/mat/fun/positive_ordered_constrain.hpp`

They use `exp(double)` without including a definition or bringing `std::exp` into scope with a using statement.  The fix is to include `<cmath>` and add a `using std::exp` in functions that use it.  

The header tests didn't catch it because there's an `exp()` definition lurking in `::` for C99 compatibility.  These have caused us enormous headaches in the past for this reason.

I found this bug when introducing `exp(const std::complex<T>&)`, which broke the header tests.   

#### Current Version:
v3.0.0

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR fixes the files that were not respecting include-what-you-use with respect to <cmath> and the lack of `using std::*`.
It also removes an include of `<math.h> `and does some minor include cleanup.

## Tests

/

## Side Effects

/

## Checklist

- [x] Math issue Fixes #1485 

- [x] Copyright holder: Rok Češnovar

- [x] the basic tests are passing

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: stan/math/fwd/scal/fun/inv_logit.hpp [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #include <stan/math/fwd/meta.hpp>
 #include <stan/math/fwd/core.hpp>
 #include <stan/math/prim/scal/fun/inv_logit.hpp>
+#include <cmath>
```
Comment (bob-carpenter) [inline stan/math/fwd/scal/fun/inv_logit.hpp]: [question]
Any idea why cpplint didn't catch this?  I thought they were supposed to catch problems with include-what-you-use from the std:: lib.  Anyway, thanks for finding and fixing all this!

File: stan/math/opencl/opencl_context.hpp [hunk 2]
```diff
@@ -107,7 +107,7 @@ class opencl_context_base {
         in_order_ = CL_TRUE;
       }
       int thread_block_size_sqrt
-          = static_cast<int>(sqrt(static_cast<double>(max_thread_block_size_)));
+          = static_cast<int>(std::sqrt(static_cast<double>(max_thread_block_size_)));
```
Comment (bob-carpenter) [inline stan/math/opencl/opencl_context.hpp]: [optional]
With an explicit qualification for `std::sqrt`, we shouldn't need the cast to `double`---the implementation of `std::sqrt` in C++11 has all the primitives as overloads.  See:  https://en.cppreference.com/w/cpp/numeric/math/sqrt

File: stan/math/prim/scal/fun/erfc.hpp [hunk 3]
```diff
@@ -21,7 +21,7 @@ inline double erfc(double x) { return std::erfc(x); }
 
 /**
  * Return the error function of the specified argument.  This
- * version is required to disambiguate <code>erf(int)</code>.
+ * version is required to disambiguate <code>erfc(int)</code>.
```
Comment (bob-carpenter) [inline stan/math/prim/scal/fun/erfc.hpp]: [comment]
Nice catch!

**Comment by bob-carpenter** [comment]
Thanks.  And feel free to merge when checks pass.

**Comment by rok-cesnovar** [comment]
I am actually just about to push some more missing <cmath> includes and then its done

File: stan/math/fwd/scal/fun/inv_logit.hpp (hunk 1 above)
Comment (rok-cesnovar) [reply stan/math/fwd/scal/fun/inv_logit.hpp]: I was amazed also, so dont have an answer. I made a simple Python script and just ran it through  `stan/math` and fixed the files the script flagged.

File: stan/math/opencl/opencl_context.hpp (hunk 2 above)
Comment (rok-cesnovar) [reply stan/math/opencl/opencl_context.hpp]: Done! Thanks.

**Comment by rok-cesnovar** [comment]
Thanks for the review! I fixed the static cast and added the rest of missing `<cmath>` includes. Will tag you if/when it passes tests to get this merged.

**Comment by rok-cesnovar** [comment]
Something weird happened here. I merged this PR, the commit is on develop (https://github.com/stan-dev/math/commit/a2ef4ff23d293c4004877fd2af4ad126f31e6ffc) but it didn't close this PR and it's now showing like it has merge conflicts.

**Comment by rok-cesnovar** [comment]
Probably related to https://www.githubstatus.com/incidents/4g79klqzy99s better leave it and wait for github to resolve it.

**Pull request merged**